core/src/main/java/io/javaoperatorsdk/jenvtest/KubeAPIServer.java

```diff
@@ -37,6 +37,11 @@ public KubeAPIServer(KubeAPIServerConfig config) {
   }
 
   public void start() {
+    startAsync();
+    waitUntilReady();
+  }
+
+  public void startAsync() {
```

Comment:
What is the point of having these 2 public methods separately added?

Reply:
I was trying to implement a workaround for this in my own code and struggling with the "all or nothing" behavior of `KubeAPIServer.start()`. If the hard-wired readiness check fails, and the caller does not want to give up yet, then they can do some like this:

```java
try {
    apiServer.start();
} catch (IOException e) {
    // insert longer delay than the polling interval used by jenvtest and try again
    Thread.sleep(...);
    apiServer.waitUntilReady();
}
```

This might succeed in this case.

Without this, the only thing we can do to overcome flakey readiness failures is invoke `KubeAPIServer.stop()` and then re-invoke `KubeAPIServer.start()`, which is likely to fail again and also takes longer.

I'm fine with backing this out, since it does not directly address this issue but is more of a general improvement.

Reply:
Pls revert, since those public methods are not called from code, not testes and/or samples show. Feel free to create a separate PR for it where we could discuss how to proceed regarding that topic. Thx!!